unittest/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+
+include(ExternalProject)
+ExternalProject_Add(gtest
+                    GIT_REPOSITORY  https://github.com/google/googletest.git
+                    GIT_TAG         release-1.10.0
+                    SOURCE_DIR      "${CMAKE_BINARY_DIR}/gtest-src"
+                    BINARY_DIR      "${CMAKE_BINARY_DIR}/gtest-build"
+                    CMAKE_ARGS      ${CMAKE_REQUEST_PIC} ${CMAKE_EXTRA_GTEST_OPTS}
+                                    -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                                    -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                                    -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+                                    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                                    -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+                                    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+                    BUILD_COMMAND   ${CMAKE_COMMAND} --build .
+                    BUILD_BYPRODUCTS <BINARY_DIR>/lib/libgtest.a
+                    INSTALL_COMMAND ""
+                    TEST_COMMAND    "")
+
+ExternalProject_Get_Property(gtest SOURCE_DIR)
+set(GTEST_INCLUDE_DIR ${SOURCE_DIR}/include)
+ExternalProject_Get_Property(gtest BINARY_DIR)
+set(GTEST_LIBRARY_PATH ${BINARY_DIR}/lib/${CMAKE_FIND_LIBRARY_PREFIXES}gtest.a)
+
+# Prevent GoogleTest from overriding our compiler/linker options
+# when building with Visual Studio
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+
+include(FindThreads)
+set(GTEST_LIBRARY GTest::GTest)
+add_library(${GTEST_LIBRARY} UNKNOWN IMPORTED)
+set_target_properties(${GTEST_LIBRARY} PROPERTIES
+        IMPORTED_LOCATION ${GTEST_LIBRARY_PATH}
+        INTERFACE_LINK_LIBRARIES Threads::Threads)
+add_dependencies(${GTEST_LIBRARY} gtest)
```

Comment:
I think you want to create an imported target here.

Reply:
Do you want to allow external GTest libraries as well?

Reply:
No. I tried that first, but that turned out too inconsistent. Some OS versions had too only a version, others would only install the sources when installing the package. It is a fairly small library and the latest (stable) version has some convenient features that we may want to use in future tests. After discussing options with @rbberger we settled for simply downloading the same version as source and compile it. Building the test tree is not going to be a prerequisite for running LAMMPS, so we can be a bit more free in how high we set the bar.